### ISSUE TITLE
Request: evaluation_tools.nwis_client CLI

### ISSUE THREAD

**Issue by jarq6c** [NOAA-OWP/evaluation_tools#35]
This is a low priority, but CLIs for different evaluation tools may be useful to non-Python users and various old wizards that like doing everything through bash and system calls. The workflows of these users might benefit from evaluation_tools features like standardized data formats, efficient data retrieval, auto-parsing, caching, canned evaluations, etc. A CLI is a way to meet them halfway.

I can imagine something like:

$ evaluation_tools nwis_client --sites 02146600 --output USGS_site_data_02146600.csv

I had a good experience using Click to implement simple CLIs for my personal workflows. I'd like to explore this more.

**Comment by aaraney** [comment]
With old wizards in mind, is the intended functionality to use a long data format as adhered to currently, or pivot output to a wide format? Just curious your thoughts on handling multiple stations.

**Comment by jarq6c** [comment]
> With old wizards in mind, is the intended functionality to use a long data format as adhered to currently, or pivot output to a wide format? Just curious your thoughts on handling multiple stations.

Ah yes, the wide format (one station per column) would more closely match the USGS rdb tab-separated output. That could be an option later. However, I suspect an output format with a fixed number of columns might be easier to use. If we assume old wizards prefer to manually parse CSV output line-by-line in C or FORTRAN, I think they will have an easer time if the number of columns is fixed.

**Comment by aaraney** [comment]
I agree, I think a long format would be best to implement from both our perspective and any perspective sorcerers.

**Comment by jarq6c** [comment]
Possible CSV output format:

```console
# NCOLS 7
# NROWS 5
# HEADR 1
#
# value_date: Datetime of measurement (UTC) (character string)
# variable: USGS variable name (character string)
# usgs_site_code: USGS Gage Site Code (character string)
# measurement_unit: Units of measurement(character string)
# value: Measurement value (character string)
# qualifiers: Qualifier string (character string)
# series: Series number in case multiple time series are returned (integer)
#
# Generated using nwis_client v1.3.0
# Retrieved 2021-02-18T13:51:02
# 
value_date,variable_name,usgs_site_code,measurement_unit,value,qualifiers,series
2019-09-29T23:00:00,streamflow,02146470,ft3/s,0.72,['A'],0
2019-09-30T00:00:00,streamflow,02146470,ft3/s,0.79,['A'],0
2019-09-30T01:00:00,streamflow,02146470,ft3/s,0.79,['A'],0
2019-09-30T02:00:00,streamflow,02146470,ft3/s,0.72,['A'],0
2019-09-30T03:00:00,streamflow,02146470,ft3/s,0.79,['A'],0
```

**Comment by aaraney** [comment]
Good addition including metadata. I was not familiar with `pd.read_csv(comment="#")` option until now. I like this functionality. With this in mind, I think including a csv writer in the `nwis_client` is a good idea to support users who may want to distribute data in a canonical `evaluation_tools` csv format? Maybe my definition of format is a little loose here, but I hope you get what I mean.
